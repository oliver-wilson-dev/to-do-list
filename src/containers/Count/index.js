import { connect } from 'react-redux';
import Count from '../../components/Count';
import { getCount } from '../../state/selectors';

const mapStateToProps = state => ({
  count: getCount(state)
});

export default connect(mapStateToProps)(Count);
