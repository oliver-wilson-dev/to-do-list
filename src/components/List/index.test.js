import React from 'react';
import { shallow } from 'enzyme';
import Count from './index';

const defaultProps = { count: 0 };

const render = overrideProps => shallow(<Count {...defaultProps} {...overrideProps} />);

describe('<Count/> component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });


  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when the count prop changes', () => {
    it('should reflect that change in the h1 text', () => {
      expect(render().setProps({ count: 123 })).toMatchSnapshot();
    });
  });
});
