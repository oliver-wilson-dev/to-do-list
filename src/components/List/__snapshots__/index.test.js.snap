// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Count/> component should render correctly 1`] = `
<h1>
  Clicked 
  0
   times.
</h1>
`;

exports[`<Count/> component when the count prop changes should reflect that change in the h1 text 1`] = `
<h1>
  Clicked 
  123
   times.
</h1>
`;
